hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/ListingBasedRollbackStrategy.java

```diff
@@ -272,17 +287,30 @@ private FileStatus[] fetchFilesFromCommitMetadata(HoodieInstant instantToRollbac
    * @param partitionPath
    * @param basePath
    * @param baseFileExtension
-   * @param fs
+   * @param storage
    * @return
    * @throws IOException
    */
-  private FileStatus[] fetchFilesFromListFiles(HoodieInstant instantToRollback, String partitionPath, String basePath,
-                                               String baseFileExtension, FileSystem fs)
-      throws IOException {
-    SerializablePathFilter pathFilter = getSerializablePathFilter(baseFileExtension, instantToRollback.getTimestamp());
-    Path[] filePaths = listFilesToBeDeleted(basePath, partitionPath);
-
-    return fs.listStatus(filePaths, pathFilter);
+  private List<StoragePathInfo> fetchFilesFromListFiles(HoodieInstant instantToRollback,
+                                                        String partitionPath,
+                                                        String basePath,
+                                                        String baseFileExtension,
+                                                        HoodieStorage storage) {
+    StoragePathFilter pathFilter = getPathFilter(baseFileExtension, instantToRollback.getTimestamp());
+    StoragePath[] filePaths = listFilesToBeDeleted(basePath, partitionPath);
+
+    return Arrays.stream(filePaths)
+        .map(path -> {
+          try {
+            return storage.listDirectEntries(path, pathFilter);
+          } catch (IOException ioe) {
+            LOG.error("Failed to get StoragePathInfo for " + path.toString(), ioe);
+          }
+          return null;
+        })
```

Comment:
Might be good to add a new API `HoodieStorage#listDirectEntries` that takes a list of paths and do this logic, since this is used multiple times.

Reply:
I added a new API `listDirectEntries(List<StoragePath>, StoragePathFilter)`